Bottom: 4bae6bb3368fc28fd56c741e8c84fb2c8c242fab
Top:    1a240e3c87522c5ea39d1484cb7b1c005aa31c61
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 19:41:25 +0000

Revert "Remove 511-character limit on configuration file lines"

This reverts commit d3830fdd06e18e05ef986210a3a9da2f49498195.

Tested-by: Kai-Martin Knaak <knaak@iqo.uni-hannover.de>


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index d35650b..2c44af6 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -475,14 +475,13 @@ static void
 config_file_read (void)
 {
   FILE *f;
-  char *buf = NULL;
-  size_t n = 0;
-  gchar *option, *arg;
+  gchar buf[512], *option, *arg;
 
   if ((f = config_file_open ("r")) == NULL)
     return;
 
-  while (getline (&buf, &n, f) != -1)
+  buf[0] = '\0';
+  while (fgets (buf, sizeof (buf), f))
     {
       if (parse_option_line (buf, &option, &arg) > 0)
 	set_config_attribute (option, arg);
@@ -490,7 +489,6 @@ config_file_read (void)
       g_free (arg);
     }
 
-  free (buf);
   fclose (f);
 }
